File: project-java-cosminpac04/motorcycleContest/src/main/java/motorcycle/service/UserService.java
package motorcycle.service;
import java.util.List;
import motorcycle.model.User;
import motorcycle.repository.UserRepository;

public class UserService {
    private final UserRepository userRepository;

    public UserService(UserRepository userRepository){
        this.userRepository = userRepository;
    }

    public void addUser(User user){
        userRepository.add(user);
    }

    public void removeUser(User user){
        userRepository.delete(user);
    }

    public void updateUser(User user, int id){
        userRepository.update(user, id);
    }

    public User findUserById(int id){
        return userRepository.findById(id);
    }

    public List<User> getAllUsers(){
        return userRepository.findAll();
    }

}

File: project-c-cosminpac04/motorcycleApp/Services/UserService.cs
using System.Collections.Generic;
using motorcycleApp.Models;
using motorcycleApp.Repositories;

namespace motorcycleApp.Services
{
    public class UserService
    {
        private readonly UserRepository _userRepository;

        public UserService(UserRepository userRepository)
        {
            _userRepository = userRepository;
        }

        public void AddUser(User user)
        {
            _userRepository.Add(user);
        }

        public void RemoveUser(int id)
        {
            var user = _userRepository.FindById(id);
            _userRepository.Delete(user);
        }

        public void UpdateUser(int id, User user)
        {
            _userRepository.Update(user);
        }

        public User GetUserById(int id)
        {
            return _userRepository.FindById(id);
        }

        public IEnumerable<User> GetAllUsers()
        {
            return _userRepository.GetAll();
        }
    }
}